trio/_sync.py

```diff
@@ -965,13 +965,11 @@ def get_nowait(self):
         task = _core.current_task()
 
         def abort_fn(_):
+            self._get_wait.pop(task, None)
```

Comment:
This is a small nitpick, but: is the `None` necessary? I think it isn't, and if it isn't then I'd prefer not to have it here, because this code is already confusing and it's better not to trick the reader into thinking we're trying to handle a case that doesn't actually exist. (Also it could potentially hide a bug.)

Other than that, looks good to me, so feel free to merge after resolving that nitpick one way or the other.